baseapp/abci.go

```diff
@@ -891,6 +891,14 @@ func (app *BaseApp) FinalizeBlock(req *abci.RequestFinalizeBlock) (*abci.Respons
 	if res != nil {
 		res.AppHash = app.workingHash()
 	}
+
+	// call the streaming service hooks with the FinalizeBlock messages
+	for _, streamingListener := range app.streamingManager.ABCIListeners {
+		if err := streamingListener.ListenFinalizeBlock(app.finalizeBlockState.ctx, *req, *res); err != nil {
+			panic(fmt.Errorf("ListenFinalizeBlock listening hook failed, height: %d, err: %w", req.Height, err))
```

Comment:
We should return an error. It will still panic then.

Reply:
What about just logging the error like in `ListenCommit`?
```go
for _, abciListener := range abciListeners {
	if err := abciListener.ListenCommit(ctx, *resp, changeSet); err != nil {
		app.logger.Error("Commit listening hook failed", "height", blockHeight, "err", err)
	}
}
```

Reply:
lets return an error instead of panicking

Reply:
Possibly, even better for consistency: https://github.com/cosmos/cosmos-sdk/blob/3853e7ff70ff68b30a20365ca95b05acfea8d950/baseapp/abci.go#L949
Or switch the other to return an error as well.

FYI: https://github.com/cosmos/cosmos-sdk/pull/14207 changed the previous panicking to logging.

Given that CometBFT will still panic at errors, maybe that was the reason.

Reply:
So, just log the error without panicking? Instrucions are a bit unclear

Reply:
I think we should stay consistent, so if we return an error, we need to update it for the commit listening hook. If we don't, then this one should as well just log.
Personally, maybe it is a bit hard to stop a node if a streaming method returns an error.

Reply:
ah sorry, was thinking wrong, we dont want to error here. There is some consistency within streamer but the streamer should not panic or halt the node in baseapp

Reply:
only log, sorry for confusion

Reply:
Fix applied [here](https://github.com/cosmos/cosmos-sdk/pull/18486/commits/e465962b66377519fca00d761b1f540e8b4663c6). Listeners will only log and never panic or return errors.